### DISCUSSION TITLE
Environments

### DISCUSSION THREAD

**Discussion by baev** [allure-framework/allure2#2036] [Ideas & Feedback]

**Comment by sseliverstov** [comment]
@baev it is really still open?

**Comment by baev** [comment]
Yep

**Comment by DITLPANW** [comment]
Hi, There is something new?

**Comment by baev** [comment]
We need to decide how to display environments in the report

**Comment by DITLPANW** [comment]
![image](https://user-images.githubusercontent.com/29203401/29624848-d4b03c04-8832-11e7-9363-82efc7a6ef06.png)
This is the way I see the failure with "XML report processing" via TeamCity

**Comment by DITLPANW** [comment]
@baev  What about the following display : 
https://github.com/allure-framework/allure1/wiki/Environment

**Comment by netmajor** [comment]
For browser environment maybe it should be placed on Test page, next to Retries and on page show Browser used by test and ability to switch to other test run in other browser.

Also on Suite page should we display on suite bar, next to suite name , icons of browser used in Suite with ability to filter Suite by browser.

**Comment by Jiff21** [comment]
Along with browser, it would be nice to be able to sort by Environment Name  (Production, Test, Staging. etc). There's no plugins for the CI tools I use so currently I think I'd have to send these to a custom server per environment.

**Comment by mfestrada** [comment]
Is this still being worked on?

**Comment by SymphonicJohn** [comment]
The (Ruby) work around I user for this is to attach an environment string in my formatter before so no two xml nodes have the same name before passing to allure:
`
class AllureCucumber::Formatter
  
  def scenario_name(keyword, name, *args)
    # Patch to append browser name to test-case
    driver_type = CAP_HASH['envName'] || "Cucumber #{Time.now.to_s}"
    scenario_name = (name.nil? || name == "") ? "Unnamed scenario" : name.gsub(/\n/, " ") + " " + driver_type
    @scenario_outline ? @scenario_outline_name = scenario_name : @tracker.scenario_name = scenario_name
  end
`

**Comment by LelantosTital** [comment]
More then a year later I now face the same issue. Is there a solution to have separate environments?
